app/Http/Controllers/GoogleAuthController.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\User;
+use Illuminate\Support\Facades\Auth;
+use Laravel\Socialite\Facades\Socialite;
+use Laravel\Socialite\Two\InvalidStateException;
+
+
+class GoogleAuthController extends Controller
+{
+    public function redirectToGoogle()
+    {
+        return Socialite::driver('google')->redirect();
+    }
+
+    public function handleGoogleCallback()
+    {
+        try {
+            $socialUser = Socialite::driver('google')->user();
+        } catch (InvalidStateException $exception) {
+
+            return redirect()->route('login')
+                ->withErrors(
+                    [
+                        'email' => [
+                            __('Google Login failed, please try again.'),
+                        ],
+                    ]
+                );
+        }
+
+
+        $user = User::where('email', $socialUser->getEmail())->first();
+
+
+        if ($user) {
+            $user->update([
+                'avatar'   => $socialUser->avatar,
+            ]);
+
+            Auth::login($user, true);
+            return redirect()->route('setup.done');    
+        }
+        
+        return redirect()->route('login')
+                ->withErrors(
+                    [
+                        'email' => [
+                            'User not found.',
```

Comment:
Do we have a translation string with something like this in it? I'm thinking `admin/users/message:user_not_found'` which you would use as `trans('admin/users/message.user_not_found')` (That message ends up saying:
> User [:id] does not exist.

You could pass the email as the ID, or just pass a blank string as the `:id` instead if you prefer. I think I'm okay with it either way. The reason I'm okay with exposing the email is, you *just* logged in to Google with this account, you know exactly what email it is - and your login was succesful.